source/auth/auth.rst

```diff
@@ -683,6 +683,176 @@ mechanism
+BSON representation and a human readable "friendly name" which is used in this document. They are as follows:
+
+==== ==================== ========== ============================================================================================================================================== 
+Name Friendly Name        Type       Description
+==== ==================== ========== ==============================================================================================================================================
+r    client nonce         BinData    32 byte cryptographically secure random number 
+p    gs2-cb-flag          32-bit int The integer representation of the ASCII charater 'n' or 'y', i.e., ``110`` or ``121``
+s    server nonce         BinData    64 bytes total. The first 32 bytes from the client nonce and the second 32 bytes from the server nonce
+h    sts hosts            string     DNS Host name of the STS service 
+a    auth_header          string     Authorization header for `AWS Signature Version 4 <https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html?shortFooter=true>`_
+d    X-AMZ-Date           string     Current date in UTC -see `AWS Signature Version 4 <https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html?shortFooter=true>`_
+t    X-AMZ-Security-Token string     Optional AWS security token
+==== ==================== ========== ============================================================================================================================================== 
+
+Drivers MUST NOT advertise support for channel binding, as the server does
+not support it and legacy servers may fail authentication if drivers advertise
+support. The client-first-message MUST set ``p`` to the integer representation 
+of the ASCII character ``n``, i.e., ``110``.
+
+Conversation
+````````````
+As an example, given an ``r`` value of "enJwWTtNSkR+WztFZCE3d1NWSiMpfU54YCgmPU5lY1Q=", a MONGODB-IAM conversation would appear as
+follows:
+
+| C: ``r=enJwWTtNSkR+WztFZCE3d1NWSiMpfU54YCgmPU5lY1Q=,p=110``
+| S: ``s=enJwWTtNSkR+WztFZCE3d1NWSiMpfU54YCgmPU5lY1RHbnN1IWy6vp7GvmtRmcGWYEtjedGEI0ZXi13r7y4V+A==,h="sts.amazonaws.com"``
+
+.. note::
+	Drivers MUST validate that the server nonce is exactly 64 bytes and the first 32 bytes are the same as the client nonce, i.e., ``r``. Drivers must 
+        also validate that the host is greater than 0 and less than or equal to 255 bytes per `RFC 1035 <https://tools.ietf.org/html/rfc1035>`_. Drivers MUST reject
+        DNS names with empty labels, e.g., "abc..def" and error on any additional fields.
+
+| C: ``a="AWS4-HMAC-SHA256 Credential=AKIAICGVLKOKZVY3X3DA/20191017/us-east-1/sts/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-mongodb-gs2-cb-flag;x-mongodb-server-nonce, Signature=6f67c4dfbc05da04ea8d571bc6f7d11833f9d1909d0068be302c5379e1867d15",d="20191017T173547Z"``
```

Comment:
The case of various placeholder names in this example message does not match with the corresponding entries in the mini-table below. I think it would be less confusing to have everything be the same case so that there is no ambiguity. 
Another option is to use `<.>` syntax to indicate that these are placeholders and then specify them in the table as already done. Right now the example mixes real values (e.g. the nonce literal) with field/entry name placeholders which doesn't seem very intuitive.

Reply:
Updated.

Reply:
@prashantmital please comment back or resolve.